userspace/libsinsp/threadinfo.h

```diff
@@ -230,8 +230,11 @@ class SINSP_PUBLIC sinsp_threadinfo
 	string m_exe; ///< argv[0] (e.g. "sshd: user@pts/4")
 	string m_exepath; ///< full executable path
 	vector<string> m_args; ///< Command line arguments (e.g. "-d1")
+	string m_args_str; // m_args combined, separated by '\0', capped to SCAP_MAX_ARGS_SIZE
 	vector<string> m_env; ///< Environment variables
+	string m_env_str; //  m_env combined, separated by '\0', capped to SCAP_MAX_ENV_SIZE
 	vector<pair<string, string>> m_cgroups; ///< subsystem-cgroup pairs
+	string m_cgroups_str; // m_cgroups combined, separated by '\0', cappped to SCAP_MAX_CGROUPS_SIZE
```

Comment:
cgroups have already a pretty big memory impact, doubling it worries me. I would test memory usage difference with an high number of threads (40k) for example.

A possible alternative on my mind can be have just strings as storage and just at `getter` time, parse them and return the expected value to the caller. It would be great to have a `string` object that can point to an external storage buffer.